pkg/nfs-checker/checker.go

```diff
@@ -83,64 +73,32 @@ func (c *checker) Write() error {
 // Check checks the directory and returns the result,
 // based on the configuration.
 func (c *checker) Check() CheckResult {
-	// list all files under this directory
 	dir := filepath.Join(c.cfg.VolumePath, c.cfg.DirName)
-	pattern := filepath.Join(dir, "*")
-
-	matches, err := c.listFilesByPattern(pattern)
-	if err != nil {
-		return CheckResult{
-			Dir:     dir,
-			Message: "failed to list files",
-			Error:   err.Error(),
-		}
-	}
-
-	// sort in order to make the output consistent
-	// between subsequent checks
-	sort.Strings(matches)
+	file := c.cfg.fileSelf()
 
 	result := CheckResult{
 		Dir: dir,
 	}
-	for _, file := range matches {
-		result.ReadIDs = append(result.ReadIDs, filepath.Base(file))
-
-		contents, err := os.ReadFile(file)
-		if err != nil {
-			result.Message = "failed"
-			result.Error = fmt.Sprintf("failed to read file %s: %s", file, err)
-			return result
-		}
-
-		if string(contents) != c.cfg.FileContents {
-			result.Message = "failed"
-			result.Error = fmt.Sprintf("file %q has unexpected contents", file)
-			return result
-		}
+
+	contents, err := os.ReadFile(file)
+	if err != nil {
+		result.Message = "failed"
+		result.Error = fmt.Sprintf("failed to read file %s: %s", file, err)
+		return result
 	}
 
-	if len(result.ReadIDs) < c.cfg.NumExpectedFiles {
+	if string(contents) != c.cfg.FileContents {
 		result.Message = "failed"
-		result.Error = fmt.Sprintf("expected %d files, but only %d files were read", c.cfg.NumExpectedFiles, len(result.ReadIDs))
+		result.Error = fmt.Sprintf("file %q has unexpected contents", file)
 		return result
 	}
 
-	result.Message = fmt.Sprintf("successfully checked directory %q with %d files", c.cfg.VolumePath, len(matches))
+	result.Message = fmt.Sprintf("correctly read/wrote on %q", c.cfg.VolumePath)
 	return result
 }
 
-// Clean cleans up the files in the directory with the TTL.
+// Clean cleans up the file that is written by the checker.
 func (c *checker) Clean() error {
-	// list all files under this directory
-	dir := filepath.Join(c.cfg.VolumePath, c.cfg.DirName)
-	pattern := filepath.Join(dir, "*")
-
-	// remove all files that are older than the TTL
-	// in order to make sure the checker only checks
-	// the latest file writes/reads from other members in the group
-	now := time.Now().UTC()
-	before := now.Add(-c.cfg.TTLToDelete.Duration)
-
-	return pkgfilecleaner.Clean(pattern, before)
+	file := c.cfg.fileSelf()
+	return os.RemoveAll(file)
```

Comment:
we still need to check if the return error is equal to `NotExist`, if so, the deletion is success.

Reply:
@eahydra https://pkg.go.dev/os#RemoveAll

> If the path does not exist, RemoveAll returns nil (no error).